### DISCUSSION TITLE
Is it appropriate to calculate SHAP on new data?

### DISCUSSION THREAD

**Discussion by kransom14** [shap/shap#2011] [Q&A]
I am using SHAP in xgboost (Tree SHAP) to calculate variable importance. I am interested in the SHAP values for new data not used in model training. I can calculate these values with the packages and the output seems appropriate (though the mean value or bias value that is being compared to is for the training data). But all the examples I have seen show SHAP being calculated for the training data only. Is it appropriate to calculate SHAP for new data? Is there any reason not to trust the output for SHAP values calculated by Tree SHAP for new data?

**Comment by jainraj** [comment]
The way I made peace with this issue was to look at the big goal of ML - We build ML models so that they would work on unseen data. We want to know why the model predicted something for that unseen data. Hence, applying SHAP for unseen data makes sense.
